subsys/bluetooth/mesh/settings.c

```diff
@@ -63,6 +63,10 @@ int bt_mesh_settings_set(settings_read_cb read_cb, void *cb_arg,
 
 static int mesh_commit(void)
 {
+	if (!atomic_test_bit(bt_mesh.flags, BT_MESH_INIT)) {
```

Comment:
Isn't `mesh_commit` just one more `pre_set` function that already has this condition?

Reply:
It is defined a little bit differently than the other ones. Not sure if it can be handled in a simmilar manner or if it needs to stay as it is.

Reply:
I see it uses another macro. Probably it is better to use the same as the others but it is out of scope of this PR.

Reply:
> Probably it is better to use the same as the others but it is out of scope of this PR.

This is intentional. E.g `BT_MESH_SETTINGS_DEFINE` doesn't have `commit` handler on purpose.